browser/ui/webui/brave_rewards_ui.cc

```diff
@@ -1004,6 +1041,237 @@ void RewardsDOMHandler::GetAdsData(const base::ListValue *args) {
   web_ui()->CallJavascriptFunctionUnsafe("brave_rewards.adsData", ads_data);
 }
 
+void RewardsDOMHandler::GetAdsHistory(const base::ListValue* args) {
+  if (!ads_service_) {
+    return;
+  }
+
+  ads_service_->GetAdsHistory(base::Bind(&RewardsDOMHandler::OnGetAdsHistory,
+                                         weak_factory_.GetWeakPtr()));
+}
+
+void RewardsDOMHandler::OnGetAdsHistory(
+    const std::map<std::string, std::vector<ads::AdsHistory>>& ads_history) {
+  if (!web_ui()->CanCallJavascript()) {
+    return;
+  }
+
+  base::Value ads_history_list(base::Value::Type::LIST);
+  int id = 0;
+
+  for (const auto& entry : ads_history) {
+    base::Value ads_history_dict(base::Value::Type::DICTIONARY);
+    ads_history_dict.SetKey("id", base::Value(std::to_string(id++)));
+    ads_history_dict.SetKey("date", base::Value(entry.first));
```

Comment:
This is actually a date string (e.g., "7/2/2019"). It's used in the WebUI to display the collapsible date headers for the ads history.

Reply:
issue with this is 7/2/2019 is likely to cause issues in different timezones, would be better to store timestamps which can be formatted

Reply:
I think we'd be OK on that front, as we're using ICU functionality to format the time via `base::TimeFormatShortDateNumeric` which should be locale-aware.

We could definitely make this a timestamp instead, it's just that it would cause some extra coordination as it would also involve changes to the related brave-ui PR.

cc: @jasonrsadler for your thoughts here as well, I'm open to either approach.

Reply:
I think it would be more performant to format in the backend. Other than that, either way should work. If you bring a timestamp to the front, we could convert before sending to UI lib. 
```
new Date(stamp).toLocaleDateString()
```
if stamp is in ms (otherwise, stamp * 1000)

Reply:
and yes `base::i18n::time_formatting::TimeFormatShortDateNumeric` uses current locale

Reply:
I would personally go with formatting at the frontend, we do this with the pending rewards panel, timestamps are just a safer option in my opinion and this allows each system to define how a date is displayed based upon visual space etc.

Reply:
Makes sense, will do.